### ISSUE TITLE
NaN values in training loss when the power is decimal/fractional but not when integers

### ISSUE THREAD

**Issue by athtareq** [lululxvi/deepxde#1177]
Hello, I'm trying to solve the following problem 

```
n=1
bta=1.5
gma=2.2
alph=3.5
k1=1.1
k2=1.3
K=3

def f(t,u,v):
  a=(K-v)**bta
  b=u**alph
  c=v**gma
  return 2+tf.sin(t)-k1*a*b+k2*c
def g(t,u,v):
  a=(K-v)**bta
  b=u**alph
  c=v**gma
  return 1+tf.cos(t)+k1*a*b-(k1+k2)*c



def ode_system(t, y):
    u, v = y[:, 0:1], y[:, 1:]
    du_t = tf.gradients(u, t)[0]
    dv_t = tf.gradients(v, t)[0]
    return [du_t - f(t,u,v), dv_t - g(t,u,v)]

def boundary(_, on_initial):
    return on_initial 


geom = dde.geometry.TimeDomain(0,2*np.pi)
bc_u = dde.PeriodicBC(geom, 0, lambda _, on_boundary: on_boundary, derivative_order=0, component=0)
bc_v = dde.PeriodicBC(geom, 0, lambda _, on_boundary: on_boundary, derivative_order=0, component=1)
layer_size = [1] + [50] * 7 + [2]
activation = "tanh"
initializer = "Glorot uniform"
net = dde.maps.FNN(layer_size, activation, initializer)

data = dde.data.PDE(geom,ode_system, [bc_u, bc_v],num_domain=300, num_boundary=2)
model = dde.Model(data, net)
model.compile("adam", lr=0.001)
losshistory, train_state = model.train(epochs=10000)
model.compile("L-BFGS-B")
losshistory, train_state = model.train()

dde.saveplot(losshistory, train_state, issave=True, isplot=True)

```

But all I get is 

![Screenshot from 2023-03-09 10-51-45](https://user-images.githubusercontent.com/61631857/223985282-2ffa71ec-57ef-4e2d-b806-005c730ecb84.png)

But if I set bta, gma and alph to integer values (like 1, 2 and 3 resp.) the model compiles normally. Anyone know the reasons why ? Thanks.

**Comment by Lastilla** [comment]
If I remember correctly, you set the wrong layer size.

**Comment by athtareq** [comment]
Hello @Lastilla, thank you for your reply. I played around with the layer size but I still get the same error. Do you have any ideas or references ? Thanks.

**Comment by lululxvi** [comment]
I guess it is due to `(K-v)**bta`, when `K-v < 0`.

**Comment by athtareq** [comment]
@lululxvi thanks, you're probably right. Although it computes fine using RK4, which is what confused me.
